### PR TITLE
feat: add support for |= &= and ^=

### LINKED ISSUE #328: Add support for `|=` `&=` and `^=`
Ref: #87

### PR DESCRIPTION
**Pull request by Alejandbel**
Closes #328

- [x] I have updated CHANGELOG.md
- [x] I have added tests to demonstrate the contribution is correctly implemented: this usually includes both positive and negative tests, showing the happy path(s) and featuring intentionally broken cases
- [x] I have run all the tests locally and no test failure was reported
- [x] I did not do unrelated and/or undiscussed refactorings

### CODE REVIEW

**Review by Gusarich — DISMISSED** [review]
LGTM, great job! Also thanks for removing these whitespaces from `grammar.ohm`, they were pissing me off, haha 😆

**Comment by byakuren-hijiri** [comment]
We should consider modifying the type checker for this case, as these operations should be applied only to numeric types: https://github.com/tact-lang/tact/issues/353.

**Comment by Gusarich** [comment]
> We should consider modifying the type checker for this case, as these operations should be applied only to numeric types: #353.

@Alejandbel you don't have to change anything in your PR for that, I've resolved this in #356

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Added
 
+- Extend assign operators with `|=`, `&=`, `^=`: PR [#350](https://github.com/tact-lang/tact/pull/350)
```
Comment (anton-trunov) [inline CHANGELOG.md]:
```suggestion
- Extend augmented assignment operators with bitwise `|=`, `&=`, `^=`: PR [#350](https://github.com/tact-lang/tact/pull/350)
```

File: src/grammar/grammar.ohm
Comment (anton-trunov) [inline src/grammar/grammar.ohm:1]: The conflicts from the whitespace changes will be a pain in the neck to resolve after the upcoming PR fixing issue #341. So, let's please remove the unrelated changes from this file

File: src/grammar/grammar.ohm [hunk 2]
```diff
@@ -102,19 +102,25 @@ Tact {
...
     StatementExpression = Expression ";"
     StatementAssign = LValue "=" Expression ";"
     StatementAugmentedAssign = StatementAugmentedAssignAdd
                              | StatementAugmentedAssignSub
                              | StatementAugmentedAssignMul
                              | StatementAugmentedAssignDiv
                              | StatementAugmentedAssignRem
+                             | StatementAugmentedAssignOr
+                             | StatementAugmentedAssignAnd
+                             | StatementAugmentedAssignXor
```
Comment (anton-trunov) [inline src/grammar/grammar.ohm]: These identifiers should have `Bitwise` in them for consistency with the rest of the grammar

**Review by anton-trunov — CHANGES_REQUESTED** [review]
Thanks! I left some minor comments. Btw, we can only merge this PR after Tact v1.3.1 is released. In the mean we will inevitably introduce some merge conflicts.

**Comment by anton-trunov** [comment]
@Alejandbel Could you please update your PR after the grammar refactoring PR #365?

**Comment by Alejandbel** [comment]
> @Alejandbel Could you please update your PR after the grammar refactoring PR #365?

Yep, sure, will update it today

**Comment by Alejandbel** [comment]
> @Alejandbel Could you please update your PR after the grammar refactoring PR #365?

Updated

**Review by anton-trunov — DISMISSED** [review]
Perfect! Thanks a lot!

**Comment by anton-trunov** [comment]
Hey @Alejandbel, your contribution qualifies for the Tact contributor [SBT](https://society.ton.org/contribute-to-tact-compiler), would you mind sharing your TON wallet?

**Comment by Alejandbel** [comment]
Sure, UQBSzBN6cnxDwDjn_IQXqgU8OJXUMcol9pxyL-yLkpKzYpKR

> Hey @Alejandbel, your contribution qualifies for the Tact contributor [SBT](https://society.ton.org/contribute-to-tact-compiler), would you mind sharing your TON wallet?

**Comment by anton-trunov** [comment]
SBT sent!

**Review by anton-trunov — APPROVED** [review]

**Pull request merged**